wire-ios-request-strategy/Sources/Request Strategies/Connection/ConnectionRequestStrategyTests.swift

```diff
@@ -247,6 +251,59 @@ class ConnectionRequestStrategyTests: MessagingTestBase {
         }
     }
 
+    func testThatOneOnOneResolverIsInvoked_WhenConnectionRequestIsAccepted() throws {
+
+        try syncMOC.performAndWait {
+            // GIVEN
+            let connection = createConnectionPayload(self.oneToOneConnection, status: .accepted)
+            let eventType = try XCTUnwrap(ZMUpdateEvent.eventTypeString(for: Payload.Connection.eventType), "eventType is nil")
+            let eventPayload = Payload.UserConnectionEvent(connection: connection, type: eventType)
+            let payloadData = try XCTUnwrap(eventPayload.payloadData(), "payloadData is nil")
+            let event = updateEvent(from: payloadData)
+
+            mockOneOnOneResolver.resolveOneOnOneConversationWithIn_MockMethod = { _, _ in
+                return OneOnOneConversationResolution.noAction
+            }
+
+            sut.oneOnOneResolutionDelay = 0.3
+
+            // WHEN
+            self.sut.processEvents([event], liveEvents: true, prefetchResult: nil)
+        }
+
+        XCTAssertTrue(waitForAllGroupsToBeEmpty(withTimeout: 0.5))
+
+        // THEN
+        XCTAssertEqual(mockOneOnOneResolver.resolveOneOnOneConversationWithIn_Invocations.count, 1)
+    }
+
```

Comment:
@johnxnguyen The only way to make that test work was to use an expectation instead of waitForAllGroupsToBeEmpty(withTimeout

Reply:
I think we can improve this test by doing something like:

- set delay to 1 second
- create 2 expectations, fulfill them both in the mock method
- call the method
- wait for the first expectation with a 0.5 timeout, asserting it fails
- assert no invocations
- wait for the second expectation with a 1 second time out, asserting it succeeds
- assert an invocation.

Reply:
Done!